src/fides/api/models/sql_models.py

```diff
@@ -401,6 +403,22 @@ class System(Base, FidesBase):
         "Cookies", back_populates="system", lazy="selectin", uselist=True, viewonly=True
     )
 
+    user_id = Column(String, nullable=True)
+
+    @property
+    def created_by(self) -> Optional[str]:
+        """Derive the username from the user_id"""
+        if not self.user_id:
+            return None
+
+        if self.user_id == CONFIG.security.oauth_root_client_id:
+            return CONFIG.security.root_username
+
+        db = Session.object_session(self)
```

Comment:
This seems to cause 500 errors when creating systems with a non-root user -  I am wondering if it is because we're using a synchronous session construct here and the ctl API uses async sessions, just a theory.  
```
2023-09-01 10:42:51 2023-09-01 15:42:51.827 | INFO     | fides.api.main:log_request:150 - Request received | {'method': 'POST', 'status_code': 500, 'handler_time': '733.93ms', 'path': '/api/v1/system/'}
2023-09-01 10:42:51 2023-09-01 15:42:51.939 | DEBUG    | fides.api.oauth.utils:has_scope_subset:298 - Auth token missing required scopes: system:read. Scopes provided: .
2023-09-01 10:42:51 2023-09-01 15:42:51.940 | DEBUG    | fides.api.db.crud:list_resource:204 - Fetching resources | {'sql_model': 'System'}
2023-09-01 10:42:51 2023-09-01 15:42:51.947 | ERROR    | fides.api.util.logger:_log_exception:31 - unhandled errors in a TaskGroup (1 sub-exception)
2023-09-01 10:42:51 Traceback (most recent call last):
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/anyio/streams/memory.py", line 97, in receive
2023-09-01 10:42:51     return self.receive_nowait()
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/anyio/streams/memory.py", line 92, in receive_nowait
2023-09-01 10:42:51     raise WouldBlock
2023-09-01 10:42:51 anyio.WouldBlock
2023-09-01 10:42:51 
2023-09-01 10:42:51 During handling of the above exception, another exception occurred:
2023-09-01 10:42:51 
2023-09-01 10:42:51 Traceback (most recent call last):
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/starlette/middleware/base.py", line 77, in call_next
2023-09-01 10:42:51     message = await recv_stream.receive()
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/anyio/streams/memory.py", line 112, in receive
2023-09-01 10:42:51     raise EndOfStream
2023-09-01 10:42:51 anyio.EndOfStream
2023-09-01 10:42:51 
2023-09-01 10:42:51 During handling of the above exception, another exception occurred:
2023-09-01 10:42:51 
2023-09-01 10:42:51 Traceback (most recent call last):
2023-09-01 10:42:51   File "<string>", line 1, in <module>
2023-09-01 10:42:51   File "/usr/local/lib/python3.10/multiprocessing/spawn.py", line 116, in spawn_main
2023-09-01 10:42:51     exitcode = _main(fd, parent_sentinel)
2023-09-01 10:42:51   File "/usr/local/lib/python3.10/multiprocessing/spawn.py", line 129, in _main
2023-09-01 10:42:51     return self._bootstrap(parent_sentinel)
2023-09-01 10:42:51   File "/usr/local/lib/python3.10/multiprocessing/process.py", line 314, in _bootstrap
2023-09-01 10:42:51     self.run()
2023-09-01 10:42:51   File "/usr/local/lib/python3.10/multiprocessing/process.py", line 108, in run
2023-09-01 10:42:51     self._target(*self._args, **self._kwargs)
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/uvicorn/_subprocess.py", line 76, in subprocess_started
2023-09-01 10:42:51     target(sockets=sockets)
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/uvicorn/server.py", line 61, in run
2023-09-01 10:42:51     return asyncio.run(self.serve(sockets=sockets))
2023-09-01 10:42:51   File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
2023-09-01 10:42:51     return loop.run_until_complete(main)
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/starlette/middleware/base.py", line 69, in coro
2023-09-01 10:42:51     await self.app(scope, receive_or_disconnect, send_no_error)
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/starlette/middleware/base.py", line 106, in __call__
2023-09-01 10:42:51     response = await self.dispatch_func(request, call_next)
2023-09-01 10:42:51 > File "/fides/src/fides/api/main.py", line 73, in dispatch_log_request
2023-09-01 10:42:51     response = await call_next(request)
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/starlette/middleware/base.py", line 80, in call_next
2023-09-01 10:42:51     raise app_exc
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/starlette/middleware/base.py", line 69, in coro
2023-09-01 10:42:51     await self.app(scope, receive_or_disconnect, send_no_error)
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/starlette/middleware/cors.py", line 84, in __call__
2023-09-01 10:42:51     await self.app(scope, receive, send)
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/starlette/middleware/base.py", line 104, in __call__
2023-09-01 10:42:51     async with anyio.create_task_group() as task_group:
2023-09-01 10:42:51   File "/opt/fides/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 664, in __aexit__
2023-09-01 10:42:51     raise BaseExceptionGroup(
2023-09-01 10:42:51 exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
```

Reply:
updating, we resolved by pulling this property out for now